projects/storefrontlib/cms-components/misc/icon/icon-loader.service.ts

```diff
@@ -99,51 +107,51 @@ export class IconLoaderService {
    * no head element available and the link must be loaded for every
    * web component.
    */
-  addLinkResource(iconType: ICON_TYPE | string): void {
-    const resource: IconConfigResource | undefined = this.findResource(
-      iconType,
-      IconResourceType.LINK
-    );
-    if (
-      resource &&
-      resource.url &&
-      !this.loadedResources.includes(resource.url)
-    ) {
+  public addLinkResource(iconType: ICON_TYPE_STRING): void {
+    const resource = this.findResource(iconType, IconResourceType.LINK);
+
+    if (resource?.url && !this.loadedResources.includes(resource.url)) {
       this.loadedResources.push(resource.url);
-      const head = this.winRef.document.getElementsByTagName('head')[0];
-      const link = this.winRef.document.createElement('link');
-      link.rel = 'stylesheet';
-      link.type = 'text/css';
-      link.href = resource.url;
-      head.appendChild(link);
+      // using DOM APIs, so need to sanitize our URLs manually
+      const sanitizedUrl = this.sanitizer.sanitize(
+        SecurityContext.URL,
+        resource.url
+      );
+      if (sanitizedUrl) {
+        const head = this.winRef.document.getElementsByTagName('head')[0];
```

Comment:
Or should we use `Renderer2` rather than using `document` directly?

Reply:
I don't know the pros and cons of one or the other.  However, using the Renderer2 would require a breaking change in the constructor to add a new dependency.